src/lib.rs

```diff
@@ -160,15 +175,79 @@ impl Platform {
                         touch.location.y as f32 / self.scale_factor as f32,
                     );
 
-                    self.raw_input.events.push(egui::Event::PointerButton {
-                        pos: pointer_pos,
-                        button: egui::PointerButton::Primary,
-                        pressed: match touch.phase {
-                            winit::event::TouchPhase::Started => true,
-                            _ => false,
+                    let device_id = match self.device_indices.get(&touch.device_id) {
+                        Some(id) => *id,
+                        None => {
+                            self.device_indices.insert(touch.device_id, self.next_device_index);
+                            self.next_device_index += 1;
+                            *self.device_indices.get(&touch.device_id).unwrap()
```

Comment:
Please don't use unwrap. Either properly throw an error or use "expect" with a proper message.

Reply:
It's an unwrap in this case because we just inserted that value in the two lines above - so we 100% know that there's no error possible. Can maybe just add a trailing comment?

Reply:
Negative. Expect should be used since we can express the intent better and provide better error messages.